cpp/include/cudf/functions.h

```diff
@@ -840,6 +840,31 @@ gdf_error gdf_group_by_count(int ncols,                    // # columns
                              gdf_column* out_col_agg,      //aggregation result
                              gdf_context* ctxt);            //struct with additional info: bool is_sorted, flag_sort_or_hash, bool flag_count_distinct
 
+/* --------------------------------------------------------------------------*/
+  /**
+   * @brief Given a set of columns, of which a subset of these are defined to be the group by columns,
+   * all columns are sorted by the group by columns and returned, along with a column containing
+   * a list of the start indices of each group
+   *
+   * @Param[in] The number of columns in the dataset
+   * @Param[in] The input columns in the dataset
+   * @Param[in] The number of columns to be grouping by
+   * @Param[in] The column indices of the input dataset that will be grouped by
+   * @Param[out] The dataset sorted by the group by columns (needs to be pre-allocated)
+   * @Param[out] A column containing the starting indices of each group. Indices based off of new sort order.
+   * @Param[in] Flag indicating if nulls are smaller (0) or larger (1) than non nulls for the sort operation
+   *
+   * @Returns gdf_error with error code on failure, otherwise GDF_SUCESS
+   */
+  /* ----------------------------------------------------------------------------*/
+gdf_error gdf_group_by_wo_aggregations(int num_data_cols,
```

Comment:
Suggest just calling it `gdf_group_by` -- the existing functions all have an aggregator suffix, so this will be different enough. 
```suggestion
gdf_error gdf_group_by(int num_data_cols,
```

Reply:
`gdf_group_by` is already taken by the new groupby API: https://github.com/rapidsai/cudf/pull/426/files#diff-e51b23654e09be9793d747c928c855d7R37

Reply:
Pandas `groupby()` returns a groupby object -- it doesn't do aggregations. I really we should rename `gdf_group_by()` to something like `gdf_groupby_aggregate()`, and call the non-aggregating one `gdf_groupby()`. Am I being too logical? :)